homeassistant/components/homewizard_energy/sensor.py

```diff
@@ -0,0 +1,226 @@
+    ),
+    SensorEntityDescription(
+        key=ATTR_TOTAL_ENERGY_EXPORT_T1_KWH,
+        name="Total Power Export T1",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
+        state_class=STATE_CLASS_TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key=ATTR_TOTAL_ENERGY_EXPORT_T2_KWH,
+        name="Total Power Export T2",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
+        state_class=STATE_CLASS_TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key=ATTR_ACTIVE_POWER_W,
+        name="Active Power",
+        native_unit_of_measurement=POWER_WATT,
+        device_class=DEVICE_CLASS_POWER,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_ACTIVE_POWER_L1_W,
+        name="Active Power L1",
+        native_unit_of_measurement=POWER_WATT,
+        device_class=DEVICE_CLASS_POWER,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_ACTIVE_POWER_L2_W,
+        name="Active Power L2",
+        native_unit_of_measurement=POWER_WATT,
+        device_class=DEVICE_CLASS_POWER,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_ACTIVE_POWER_L3_W,
+        name="Active Power L3",
+        native_unit_of_measurement=POWER_WATT,
+        device_class=DEVICE_CLASS_POWER,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_TOTAL_GAS_M3,
+        name="Total Gas",
+        native_unit_of_measurement=VOLUME_CUBIC_METERS,
+        device_class=DEVICE_CLASS_GAS,
+        state_class=STATE_CLASS_TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key=ATTR_GAS_TIMESTAMP,
+        name="Gas Timestamp",
+        device_class=DEVICE_CLASS_TIMESTAMP,
+    ),
+)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Config entry example."""
+    energy_api = hass.data[DOMAIN][entry.data["unique_id"]][CONF_API]
+    coordinator = hass.data[DOMAIN][entry.data["unique_id"]][COORDINATOR]
+
+    entities = []
+    for description in SENSORS:
+        if description.key in energy_api.data.available_datapoints:
+            entities.append(HWEnergySensor(coordinator, entry.data, description))
+    async_add_entities(entities, update_before_add=True)
+
+    return True
+
+
+class HWEnergySensor(CoordinatorEntity, SensorEntity):
+    """Representation of a HomeWizard Energy Sensor."""
+
+    unique_id = None
+    name = None
+
+    def __init__(self, coordinator, entry_data, description):
+        """Initialize Sensor Domain."""
+
+        super().__init__(coordinator)
+        self.entity_description = description
+        self.coordinator = coordinator
+        self.entry_data = entry_data
+
+        # Config attributes.
+        self.name = "{} {}".format(entry_data["custom_name"], description.name)
```

Comment:
Assign to `self._attr_name` to have the name property pick it up. 

Custom name should be dropped, just use the config entry title.

Reply:
Dropped custom name with https://github.com/home-assistant/core/pull/55812/commits/79f574f2d05a226bddeb6ec3016f667565132e72. But I can't get the name to work when I use `self._attr_name`. It just uses an 'readable' unique id.

<img width="1060" alt="Screenshot 2021-12-06 at 21 16 39" src="https://user-images.githubusercontent.com/74970928/144916013-9eba0fc5-0530-40b2-a40a-1e314a7a335f.png">
.

Reply:
https://github.com/home-assistant/core/pull/55812/commits/6fba568fdf44bd1c329f30b57099aae373a25ef4